binary_heap.py

```diff
@@ -0,0 +1,53 @@
+# _*_ utf=8 _*_
+
+
+class BinHeap(object):
+    """Implement a Binary-Heap."""
+
+    def __init__(self):
+        """Initialize the Binary-Heap."""
+        self.heap = []
+        self.size = 0
+
+    def push(self, value):
+        """Push new value into Binary-Heap and adjusts order."""
+        self.heap.append(value)
+        self.size += 1
+        self._child_vs_parent(self.size - 1)
+        print(self.heap)
+        return self.heap
+
+    def _child_vs_parent(self, i):
+        """Compare the child to the parent."""
+        while (i - 1) // 2 >= 0:
+            if self.heap[(i - 1) // 2] < self.heap[i]:
+                self.heap[(i - 1) // 2], self.heap[i] = self.heap[i], self.heap[(i - 1) // 2]
```

Comment:
I agree with lines like this it's hard to keep readability if you split the line. It is possible to do if you're working in a codebase that is strict with PEP8:

```python
                (self.heap[(i - 1) // 2],
                 self.heap[i]) = self.heap[i], self.heap[(i - 1) // 2]
```

Or, you could break this out into some kind of `swap()` function. Also you can silence the linter `# NOQA`, but be careful with that. It can become a bad habit:

```python
self.heap[(i - 1) // 2], self.heap[i] = self.heap[i], self.heap[(i - 1) // 2]  # NOQA
```